pkg/scheduler/core/equivalence/eqivalence.go

```diff
@@ -33,22 +33,146 @@ import (
 	"github.com/golang/glog"
 )
 
-// Cache saves and reuses the output of predicate functions. Use RunPredicate to
+// nodeMap stores a *Cache for each node.
+type nodeMap map[string]*NodeCache
+
+// Cache is a thread safe nodeMap
```

Comment:
Because the Cache type is the main entry point of this package, I think we should keep a lot of the previous type comment with it. i.e. the fact that it saves and reuses predicate results, and the explanation of node/predicate/class heirarchy. (Those comments were moved to the NodeCache type, but I think it's okay if we repeat ourselves a bit.)


Also, this is the second significant API change I've seen that is driven by performance needs. I think we should add to the comment on this type that users are expected to precompute the equivalence.Class value and pre-lookup the NodeCache within one scheduling cycle.

Reply:
Good point, I will clarify these with more comments.